src/main/java/core/org/akaza/openclinica/dao/hibernate/EventDefinitionCrfDao.java

```diff
@@ -55,8 +55,8 @@
 
     @SuppressWarnings("unchecked")
     public EventDefinitionCrf findByStudyEventDefinitionIdAndCRFIdAndStudyId(Integer studyEventDefinitionId, Integer crfId, Integer studyId) {
-        String query = "from " + getDomainClassName() + " do where do.studyEventDefinition.studyEventDefinitionId = :studyeventdefid "
-                + " and do.study.studyId = :studyid and do.crf.crfId = :crfid";
+        String query = "select do from " + getDomainClassName() + " do,Study s where do.studyEventDefinition.studyEventDefinitionId = :studyeventdefid "
```

Comment:
try using "findAllByDefinition" in EventDefinitionCRFDAO instead of changing this query as it may break the existing methods.

Reply:
Done---
after more discussion, we decide to create a new method in hibernate DAO